src/network/network_survey.cpp

```diff
@@ -0,0 +1,409 @@
+ */
+static void SurveyFont(nlohmann::json &survey)
+{
+	survey["small"] = FontCache::Get(FS_SMALL)->GetFontName();
+	survey["medium"] = FontCache::Get(FS_NORMAL)->GetFontName();
+	survey["large"] = FontCache::Get(FS_LARGE)->GetFontName();
+	survey["mono"] = FontCache::Get(FS_MONO)->GetFontName();
+}
+
+/**
+ * Convert company information to JSON.
+ *
+ * @param survey The JSON object.
+ */
+static void SurveyCompanies(nlohmann::json &survey)
+{
+	for (const Company *c : Company::Iterate()) {
+		auto &company = survey[std::to_string(c->index)];
+		if (c->ai_info == nullptr) {
+			company["type"] = "human";
+		} else {
+			company["type"] = "ai";
+			company["script"] = fmt::format("{}.{}", c->ai_info->GetName(), c->ai_info->GetVersion());
+		}
+
+		for (VehicleType type = VEH_BEGIN; type < VEH_COMPANY_END; type++) {
+			uint amount = c->group_all[type].num_vehicle;
+			company["vehicles"][_vehicle_type_to_string[type]] = amount;
+		}
+
+		company["infrastructure"]["road"] = c->infrastructure.GetRoadTotal();
+		company["infrastructure"]["tram"] = c->infrastructure.GetTramTotal();
+		company["infrastructure"]["rail"] = c->infrastructure.GetRailTotal();
+		company["infrastructure"]["signal"] = c->infrastructure.signal;
+		company["infrastructure"]["water"] = c->infrastructure.water;
+		company["infrastructure"]["station"] = c->infrastructure.station;
+		company["infrastructure"]["airport"] = c->infrastructure.airport;
+	}
+}
+
+/**
+ * Convert GRF information to JSON.
+ *
+ * @param survey The JSON object.
+ */
+static void SurveyGrfs(nlohmann::json &survey)
+{
+	for (GRFConfig *c = _grfconfig; c != nullptr; c = c->next) {
+		auto grfid = fmt::format("{:08x}", BSWAP32(c->ident.grfid));
+		auto &grf = survey[grfid];
+
+		std::string md5sum;
+		for (int i = 0; i < 16; i++) {
+			md5sum += fmt::format("{:02x}", c->ident.md5sum[i]);
+		}
+
+		grf["md5sum"] = md5sum;
+		grf["status"] = c->status;
+
+		if ((c->palette & GRFP_GRF_MASK) == GRFP_GRF_UNSET) grf["palette"] = "unset";
+		if ((c->palette & GRFP_GRF_MASK) == GRFP_GRF_DOS) grf["palette"] = "dos";
+		if ((c->palette & GRFP_GRF_MASK) == GRFP_GRF_WINDOWS) grf["palette"] = "windows";
+		if ((c->palette & GRFP_GRF_MASK) == GRFP_GRF_ANY) grf["palette"] = "any";
+
+		if ((c->palette & GRFP_BLT_MASK) == GRFP_BLT_UNSET) grf["blitter"] = "unset";
+		if ((c->palette & GRFP_BLT_MASK) == GRFP_BLT_32BPP) grf["blitter"] = "32bpp";
+
+		std::vector<std::string> flags;
+		if (HasBit(c->flags, GCF_SYSTEM)) flags.push_back("system");
+		if (HasBit(c->flags, GCF_UNSAFE)) flags.push_back("unsafe");
+		if (HasBit(c->flags, GCF_STATIC)) flags.push_back("static");
+		if (HasBit(c->flags, GCF_COMPATIBLE)) flags.push_back("compatible");
+		if (HasBit(c->flags, GCF_COPY)) flags.push_back("copy");
+		if (HasBit(c->flags, GCF_INIT_ONLY)) flags.push_back("init_only");
+		if (HasBit(c->flags, GCF_RESERVED)) flags.push_back("reserved");
+		if (HasBit(c->flags, GCF_INVALID)) flags.push_back("invalid");
+		grf["flags"] = flags;
```

Comment:
I think the only flag that makes sense in the survey is `GCF_STATIC`. The rest is internal state during NewGRF initialisation.
```suggestion
		grf["is_static"] = HasBit(c->flags, GCF_STATIC);
```